include/exadg/matrix_free/categorization.h

```diff
@@ -50,8 +50,12 @@ do_cell_based_loops(dealii::Triangulation<dim> const & tria,
   else
     data.cell_vectorization_category.resize(tria.n_active_cells());
 
+  unsigned int const n_faces_per_cell = tria.get_reference_cells()[0].n_faces();
+  AssertThrow(tria.get_reference_cells().size() == 1,
+              dealii::ExcMessage("No mixed meshes allowed."));
```

Comment:
I do not understand this error message. Didn't we say that pure simplex meshes do not allow cell-based face loops (which is the content of the present function)?

Reply:
No this is to check if we have multiple reference cells in our triangulation, i.e. mixed mesh. If we don't check this we might end up having a wrong `n_faces_per_cell`.

Reply:
Then I would write the Assert before the line with `n_faces_per_cell`.

Reply:
done.